app/interactors/notes/ratings/destroy.rb

```diff
@@ -0,0 +1,24 @@
+module Notes
+  module Ratings
+    class Destroy
+      include Interactor
+
+      delegate :rating, to: :context
+
+      after :update_note_rating
+
+      def call
+        rating.destroy
+      end
+
+      private
+
+      def update_note_rating
+        note = rating.note
```

Comment:
I think that you can save the average rating in a note using the ratings table. In this case, you can remove the calculation of the average rating from the decorator. 

```ruby
note.update(average_rating: note.ratings.average(:value).round(1))
```